google/ggrc-core
# GGRC-3668 Use new API for updating GCA

# Issue description

GCA are described in cacheable models as `custom_attribute_values: CMS.Models.CustomAttributeValue.stubs`. These rows should be removed and new GCA API will be in use.

# Steps to test the changes

1. Create any new custom attributable object (for example, Control) with filled ca values;
2. Open dev tools:
![screenshot from 2018-01-12 18-05-40](https://user-images.githubusercontent.com/13063442/34881047-79d630da-f7c3-11e7-9309-e78125507ef4.png)

`custom_attributes` field is the old way to create new ca values.
We should use `custom_attribute_values` instead.

# Solution description

The app has a lot of places where we manipulate with CA differently. The solution provides the common way to manipulate CA for each Cacheable custom attributable instance.
We define new concepts, which will describe the work with CA.

1. **CustomAttributeObject**

It's way to connect values of `custom_attribute_values` and `custom_attribute_definitions` in the one instance. With help it, we can manipulate with CA more conveniently and use it inside components without call `ca utils` methods.

In the future it can be applied for LCA:
Before this moment there were a lot of custom attribute utils which transformed `custom_attribute_values` from one form to another and back to original form. It led to creation extra data which we sent to the server. `CaObject` will help us to prevent it.

![screenshot from 2018-01-12 18-29-05](https://user-images.githubusercontent.com/13063442/34882114-90bfa81e-f7c6-11e7-8191-fe2f20590c3f.png)

`def`, `errorsMap` and so on - extra data for the server 

**Read docs inside code for more details.**

2. **CustomAttributeAccess**

The instance which help to build, read and write `CaObject`s for instance. With help it we can easy remove old `custom_attributes` way.

**Read docs inside code for more details.**

3. **StateValidator**

In a few words, it's an object which generates some state object based on functions.
They have name `Validation Actions`.
For GCA they can generate state for check on non-emty mandatory field.
For LCA  they can generate another logic.
This is very convenient thing for the future for LCA because now validations is set inside one component and is quite bulky.

 **Read docs inside code for more details.**

# Sanity checklist




# PR Review checklist
